src/Features/Core/Portable/Copilot/ImplementationDetails.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.Copilot;
+
+internal sealed class ImplementationDetails
+{
+    public required string Message { get; init; }
+
+    public SyntaxNode? ReplacementNode { get; init; }
+}
```

Comment:
docuemnt this type.  especially about the relationship between message and replacementnode.  note: this should be an either/or type.  one value should always be null and the other should always be non-null.